ksql-engine/src/main/java/io/confluent/ksql/services/KafkaTopicClientImpl.java

```diff
@@ -104,9 +114,29 @@ public void createTopic(
     }
   }
 
+  /**
+   * We need this method because {@link AdminClient#createTopics(Collection)} does not allow
+   * you to pass in only partitions. Instead, we determine the default number from the cluster
+   * config and then pass that value back.
+   *
+   * @return the default broker configuration
+   */
+  private short getDefaultClusterReplication() {
+    try {
+      final String defaultReplication = getConfig(adminClient)
```

Comment:
Consider a retry loop...

Reply:
Yes. I will do this in another PR today since our current RetryUtil only works with Runnables. I'll replace our retryUtil with https://github.com/jhalterman/failsafe or a similar alternative.